tests/pr_changed_files/test_helpers.sh

```diff
@@ -61,16 +61,40 @@ cleanup_git_repo() {
     rm -rf "$1"
 }
 
-# Helper to assert expected output
-assert_output() {
-    local expected="$1"
-    local actual="$2"
+# Helper to assert both return code and output
+# Arguments:
+#   $1 - Actual return code
+#   $2 - Expected return code
+#   $3 - Actual output
+#   $4 - Expected output
+#   $5 - Optional message to display with the assertion result
+assert_result() {
+    local actual_code="$1"
+    local expected_code="$2"
+    local actual_output="$3"
+    local expected_output="$4"
+    local message="$5"
+
+    assert_equal "$actual_code" "$expected_code" "Exit code - $message"
+    if [[ -n "$expected_output" ]]; then
+        assert_equal "$actual_output" "$expected_output" "Output - $message"
+    fi
```

Comment:
This just occurred to me: what if we expect an empty output… but we end up getting an unexpected one? The `if [[ -n "$expected_output" ]]` here will skip calling `assert_equal` and we won't detect that the `"$actual_output"` is not empty when we expected it to be… 🤔 

So shouldn't we just not have that `if` at all and always call `assert_equal` in all cases?

```suggestion
    assert_equal "$actual_output" "$expected_output" "Output - $message"
```

Reply:
Hmm good point, agreed!